### DISCUSSION TITLE
Getting permission denied when downloading files.

### DISCUSSION THREAD

**Discussion by bobloadmire** [nextcloud/all-in-one#4121] [Questions] [answered]
I checked my storage drives and I can read and write fine from the drives, but for some reason AIO just started to throw this error when I attempt to download anyfiles. Seems out of the blue

> <html>
<body>

Error | index | OCP\Files\GenericFileException: |   | 2024-01-26T11:48:48-0800
-- | -- | -- | -- | --
Error | PHP | file_get_contents(/mnt/ncdata/appdata_ocza6aflhwfg/theming/global/0/favIcon-settings#0082c9): Failed to open stream: Permission denied at /var/www/html/lib/private/Files/Storage/Local.php#320

</body>
</html>

**Comment by szaimen** [comment]
Hi, can you run the following commands ans check if that fixes things?
```
docker exec -it nextcloud-aio-nextcloud chown 33:0 -R /mnt/ncdata/appdata_ocza6aflhwfg/theming
docker exec -it nextcloud-aio-nextcloud chmod 770 -R /mnt/ncdata/appdata_ocza6aflhwfg/theming
docker exec --user www-data -it nextcloud-aio-nextcloud php occ files:scan-app-data theming
```

**Comment by bobloadmire** [comment]
Thanks for the reply. Which container should I run this in? Master?

**Reply by szaimen** [reply]
I corrected the commands. They shouls now be able to be run in cmd

**Reply by bobloadmire** [reply]
Thanks, just to confirm, I'm running docker desktop on windows, but I should run this in a windows elevated cmd?

**Reply by szaimen** [reply]
Normal cmd should be sufficient

**Comment by bobloadmire** [comment]
ok so now I can log in and lsit files fine in next cloud, but when I try and download i get "Internal Server Error
The server was unable to complete your request.

If this happens again, please send the technical details below to the server administrator.

More details can be found in the server log.

Technical details
Remote Address: 172.19.0.1
Request ID: ELhzOjEuwIvl8eVAoLGz" on web.


Logs

> <html>
<body>

Error | no app in context | RuntimeException: Unable to open file handler to create certificate bundle "/files_external/rootcerts.crt.tmp5543415301". |   | 2024-01-28T18:52:02-0800
-- | -- | -- | -- | --
Error | PHP | opendir(/mnt/ncdata/bobloadmire/cache): Failed to open directory: Permission denied at /var/www/html/lib/private/Files/Storage/Local.php#157 |   | 2024-01-28T18:50:14-0800
Error | PHP | opendir(/mnt/ncdata/bobloadmire/cache): Failed to open directory: Permission denied at /var/www/html/lib/private/Files/Storage/Local.php#157 |   | 2024-01-28T18:50:13-0800
Error | webdav | Sabre\DAV\Exception\ServiceUnavailable: Could not open file |   | 2024-01-28T18:50:07-0800
Error | webdav | Sabre\DAV\Exception\ServiceUnavailable: Could not open file |   | 2024-01-28T18:49:47-0800
Error | PHP | opendir(/mnt/ncdata/bobloadmire/cache): Failed to open directory: Permission denied at /var/www/html/lib/private/Files/Storage/Local.php#157 |   | 2024-01-28T18:49:17-0800
Error | PHP | opendir(/mnt/ncdata/bobloadmire/cache): Failed to open directory: Permission denied at /var/www/html/lib/private/Files/Storage/Local.php#157 |   | 2024-01-28T18:49:17-0800
Error | PHP | opendir(/mnt/ncdata/bobloadmire/cache): Failed to open directory: Permission denied at /var/www/html/lib/private/Files/Storage/Local.php#157 |   | 2024-01-28T18:49:16-0800
Error | PHP | opendir(/mnt/ncdata/bobloadmire/cache): Failed to open directory: Permission denied at /var/www/html/lib/private/Files/Storage/Local.php#157 |   | 2024-01-28T18:48:33-0800
Error | PHP | opendir(/mnt/ncdata/bobloadmire/cache): Failed to open directory: Permission denied at /var/www/html/lib/private/Files/Storage/Local.php#157 |   | 2024-01-28T18:48:32-0800
Error | PHP | opendir(/mnt/ncdata/bobloadmire/cache): Failed to open directory: Permission denied at /var/www/html/lib/private/Files/Storage/Local.php#157 |   | 2024-01-28T18:48:24-0800
Error | PHP | opendir(/mnt/ncdata/bobloadmire/cache): Failed to open directory: Permission denied at /var/www/html/lib/private/Files/Storage/Local.php#157 |   | 2024-01-28T18:48:23-0800
Error | no app in context | RuntimeException: Unable to open file handler to create certificate bundle "/files_external/rootcerts.crt.tmp9556211851". |   | 2024-01-28T18:48:07-0800
Error | no app in context | RuntimeException: Unable to open file handler to create certificate bundle "/files_external/rootcerts.crt.tmp7942193540". |   | 2024-01-28T18:47:55-0800
Error | PHP | opendir(/mnt/ncdata/bobloadmire/cache): Failed to open directory: Permission denied at /var/www/html/lib/private/Files/Storage/Local.php#157 |   | 2024-01-27T16:24:43-0800
Error | PHP | opendir(/mnt/ncdata/bobloadmire/cache): Failed to open directory: Permission denied at /var/www/html/lib/private/Files/Storage/Local.php#157 |   | 2024-01-27T16:24:43-0800
Error | PHP | opendir(/mnt/ncdata/bobloadmire/cache): Failed to open directory: Permission denied at /var/www/html/lib/private/Files/Storage/Local.php#157 |   | 2024-01-27T16:24:36-0800
Error | PHP | opendir(/mnt/ncdata/bobloadmire/cache): Failed to open directory: Permission denied at /var/www/html/lib/private/Files/Storage/Local.php#157 |   | 2024-01-27T16:24:35-0800
Error | PHP | opendir(/mnt/ncdata/bobloadmire/cache): Failed to open directory: Permission denied at /var/www/html/lib/private/Files/Storage/Local.php#157 |   | 2024-01-27T16:05:38-0800
Error | PHP | opendir(/mnt/ncdata/bobloadmire/cache): Failed to open directory: Permission denied at /var/www/html/lib/private/Files/Storage/Local.php#157 |   | 2024-01-27T16:05:37-0800
Error | PHP | opendir(/mnt/ncdata/bobloadmire/cache): Failed to open directory: Permission denied at /var/www/html/lib/private/Files/Storage/Local.php#157 |   | 2024-01-27T16:05:30-0800
Error | PHP | opendir(/mnt/ncdata/bobloadmire/cache): Failed to open directory: Permission denied at /var/www/html/lib/private/Files/Storage/Local.php#157 |   | 2024-01-27T16:05:29-0800
Error | PHP | opendir(/mnt/ncdata/bobloadmire/cache): Failed to open directory: Permission denied at /var/www/html/lib/private/Files/Storage/Local.php#157 |   | 2024-01-27T15:50:20-0800
Error | PHP | opendir(/mnt/ncdata/bobloadmire/cache): Failed to open directory: Permission denied at /var/www/html/lib/private/Files/Storage/Local.php#157 |   | 2024-01-27T15:50:20-0800
Error | PHP | opendir(/mnt/ncdata/bobloadmire/cache): Failed to open directory: Permission denied at /var/www/html/lib/private/Files/Storage/Local.php#157 |   | 2024-01-27T15:50:14-0800
Error | PHP | opendir(/mnt/ncdata/bobloadmire/cache): Failed to open directory: Permission denied at /var/www/html/lib/private/Files/Storage/Local.php#157 |   | 2024-01-27T15:50:13-0800
Error | PHP | opendir(/mnt/ncdata/bobloadmire/cache): Failed to open directory: Permission denied at /var/www/html/lib/private/Files/Storage/Local.php#157 |   | 2024-01-27T15:34:20-0800
Error | PHP | opendir(/mnt/ncdata/bobloadmire/cache): Failed to open directory: Permission denied at /var/www/html/lib/private/Files/Storage/Local.php#157 |   | 2024-01-27T15:34:19-0800
Error | PHP | opendir(/mnt/ncdata/bobloadmire/cache): Failed to open directory: Permission denied at /var/www/html/lib/private/Files/Storage/Local.php#157 |   | 2024-01-27T15:34:12-0800
Error | PHP | opendir(/mnt/ncdata/bobloadmire/cache): Failed to open directory: Permission denied at /var/www/html/lib/private/Files/Storage/Local.php#157 |   | 2024-01-27T15:34:11-0800
Error | PHP | opendir(/mnt/ncdata/bobloadmire/cache): Failed to open directory: Permission denied at /var/www/html/lib/private/Files/Storage/Local.php#157 |   | 2024-01-27T15:19:19-0800
Error | PHP | opendir(/mnt/ncdata/bobloadmire/cache): Failed to open directory: Permission denied at /var/www/html/lib/private/Files/Storage/Local.php#157 |   | 2024-01-27T15:19:18-0800
Error | PHP | opendir(/mnt/ncdata/bobloadmire/cache): Failed to open directory: Permission denied at /var/www/html/lib/private/Files/Storage/Local.php#157 |   | 2024-01-27T15:19:11-0800
Error | PHP | opendir(/mnt/ncdata/bobloadmire/cache): Failed to open directory: Permission denied at /var/www/html/lib/private/Files/Storage/Local.php#157 |   | 2024-01-27T15:19:11-0800
Error | PHP | opendir(/mnt/ncdata/bobloadmire/cache): Failed to open directory: Permission denied at /var/www/html/lib/private/Files/Storage/Local.php#157 |   | 2024-01-27T15:09:15-0800
Error | PHP | opendir(/mnt/ncdata/bobloadmire/cache): Failed to open directory: Permission denied at /var/www/html/lib/private/Files/Storage/Local.php#157 |   | 2024-01-27T15:09:15-0800
Error | PHP | opendir(/mnt/ncdata/bobloadmire/cache): Failed to open directory: Permission denied at /var/www/html/lib/private/Files/Storage/Local.php#157 |   | 2024-01-27T14:58:59-0800
Error | PHP | opendir(/mnt/ncdata/bobloadmire/cache): Failed to open directory: Permission denied at /var/www/html/lib/private/Files/Storage/Local.php#157 |   | 2024-01-27T14:58:58-0800
Error | PHP | opendir(/mnt/ncdata/bobloadmire/cache): Failed to open directory: Permission denied at /var/www/html/lib/private/Files/Storage/Local.php#157 |   | 2024-01-27T14:58:51-0800
Error | PHP | opendir(/mnt/ncdata/bobloadmire/cache): Failed to open directory: Permission denied at /var/www/html/lib/private/Files/Storage/Local.php#157 |   | 2024-01-27T14:58:50-0800
Error | PHP | opendir(/mnt/ncdata/bobloadmire/cache): Failed to open directory: Permission denied at /var/www/html/lib/private/Files/Storage/Local.php#157


</body>
</html>

**Comment by szaimen** [comment, marked as answer]
I see. please try the following:
```
docker exec -it nextcloud-aio-nextcloud chown 33:0 -R /mnt/ncdata/
docker exec -it nextcloud-aio-nextcloud chmod 770 -R /mnt/ncdata/
docker exec --user www-data -it nextcloud-aio-nextcloud php occ files:scan-app-data -vvv
```

**Comment by bobloadmire** [comment]
so for whatever reason the commands don't seem to run, or hang? image attached.

https://i.imgur.com/Z8tffwM.png

**Reply by szaimen** [reply]
I guess it just takes a while since it adjust the permissions for all files in that folder...

**Reply by bobloadmire** [reply]
oh wow you're right, it took about 10 minutes on a ssd. Never seen chown or chmod take that long. it's running the file scan now, thanks.

**Comment by bobloadmire** [comment]
> I see. please try the following:
> 
> ```
> docker exec -it nextcloud-aio-nextcloud chown 33:0 -R /mnt/ncdata/
> docker exec -it nextcloud-aio-nextcloud chmod 770 -R /mnt/ncdata/
> docker exec --user www-data -it nextcloud-aio-nextcloud php occ files:scan-app-data -vvv
> ```


ok this appears to have fixed everything. Just for people googling and my own curiosity, do you know how the permissions got changed and what we can do to avoid this happening? I certainly haven't been mucking about with folder permissions. Thanks again for all of your help!

**Reply by szaimen** [reply]
> do you know how the permissions got changed?

No, maybe a docker desktop upsate changed the permissions? Nothing we can fix easily in AIO I fear...
